### PR TITLE
Fix nan for valid parameters in jax implementation of Multinomial

### LINKED ISSUE #1327: Nans in JAX multinomial dispatch
### Description

PyMC tests are failing: https://github.com/pymc-devs/pymc/pull/7740

Reproducible example:
```python
import pytensor.tensor as pt

p = pt.eye(3)
rv = pt.random.multinomial(n=5, p=p)
rv.eval(mode="JAX")
# Array([[ 5., nan, nan],
#        [ 0.,  5.,  0.],
#        [ 0.,  0.,  5.]], dtype=float64)
```

I guess it could be a problem with binomial with p=0?

CC @educhesne

### PR DESCRIPTION
**Pull request by ricardoV94**
Closes #1327 

----
📚 Documentation preview 📚: https://pytensor--1328.org.readthedocs.build/en/1328/

### CODE REVIEW

**Comment by educhesne** [comment]
Your fix solves the problem when p is valid (the sum of p is 1), but if not you may have `rho==0` without `p==0`; otherwise it is possible to do:
`samples = jax.random.binomial(rng, s, jnp.where((rho == 0) & (p == 0), 0, p / rho))`

**Comment by ricardoV94** [comment]
Ah the problem is the division by zero, yes that's better. I guess I should also test n=0 multinomial

**Comment by ricardoV94** [comment]
> Your fix solves the problem when p is valid (the sum of p is 1), but if not you may have `rho==0` without `p==0`; otherwise it is possible to do: `samples = jax.random.binomial(rng, s, jnp.where(rho == 0, 0, p / rho))`

Actually I don't think I want to mask that, if `p` adds to more than 1, then it's fine to get `nan`s. In numpy it would raise an error, but jax doesn't have runtime errors.

**Review by jessegrabowski — APPROVED** [review]
